### DISCUSSION TITLE
[V5] Ace Command File System Management

### DISCUSSION THREAD

**Discussion by Spedwards** [adonisjs/core#2360] [Help]
I'm attempting to create an ace command which would create files, however I'm having difficulties. I'm following V4's documentation as V5's doesn't exist for Ace's commands yet.

V4's documentation mentions file management API including methods such as `writeFile`, `ensureFile`, `ensureDir`, etc. It's not clear in V4's documentation where these are run but it makes it seem as though it belongs the `run` method. This doesn't exist in V5 at all by the looks of it or it has been renamed.

My question is, how can I interact with the file system in an Ace command? Do I have to use the `fs` module?

**Comment by thetutlage** [comment]
I recommend using the `fs-extra` module directly to achieve that. If you want to generate files from a stub template, then you can use the ace commands generator API. Reference code https://github.com/adonisjs/assembler/blob/develop/commands/Make/Base.ts#L75-L83
